### PR TITLE
Data defined properties for font marker Fix #602

### LINKED ISSUE #602: Data defined properties for font marker
https://github.com/qgis/QGIS/commit/1e3b820da4a7fcf189e2f3be4b3df7893af80296
Author: Nyall Dawson
Date:   Fri May 22 01:03:52 2015 +1000

### PR DESCRIPTION
**Pull request by yjacolin**
* Add a section into `Special tools`
* Not removing the explanation in other part (specifically for size assistant
and composition)

This is a first move to follow #921.

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -196,6 +196,33 @@ that the color picker is OS dependent and is currently not supported by OSX.
...
 
+.. _data-defined:
+
+Data-defined setup
+-------------------
+
+In some part of the vector layer properties you can find an icon |dataDefined|
+:sup:`data defined ovveride`. This tool allows you to link the parameter
+besides to a field in the attribute table and, so, to change the value of the
+parameterin a dynamic way.
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: parameter in

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -196,6 +196,33 @@ that the color picker is OS dependent and is currently not supported by OSX.
 
    Quick color selector menu
 
+.. _data-defined:
+
+Data-defined setup
+-------------------
+
+In some part of the vector layer properties you can find an icon |dataDefined|
+:sup:`data defined ovveride`. This tool allows you to link the parameter
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: override

**Comment by yjacolin** [comment]
@DelazJ can I merge?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -196,6 +196,33 @@ that the color picker is OS dependent and is currently not supported by OSX.
...
+* an entry to list the :guilabel:`Variable` available,
+* :guilabel:`Edit ...` button which open the expression window,
+* :guilabel:`Paste` button.
+
+When the data-defined is setup correctly the icon is yellow, if it is broken,
+the icon is red.
+
+Parameters that can be used with data-defined tools:
+
+* Some style parameters (font marker for instance)
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: I don't think there's a particular reason to only mention "font marker". imo, #602 is more about data-defined override option (as treated in this PR) than font markers (which was just a missing feature in the long road to data-defined everything). almost ALL parameters in the style/symbol dialog are indeed overridable.
Would say "style and symbols parameters". we could also cite some of the options concerned (no strong opinion on this)...

That said, Font markers and all other symbol layer types may need to be updated at http://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/style_library.html#marker-symbols and below

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -196,6 +196,33 @@ that the color picker is OS dependent and is currently not supported by OSX.
...
+parameter in a dynamic way.
+
+The dropdown list shows:
+
+* a :guilabel:`Description ...` that indicates if it is enabled, which input
+  expected, valid input type and the current definition,
+* an entry to list the :guilabel:`Field type` available,
+* an entry to list the :guilabel:`Variable` available,
+* :guilabel:`Edit ...` button which open the expression window,
+* :guilabel:`Paste` button.
```
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: If a expression have been used there will be also Copy and Clear options.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 3 above)
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: @DelazJ I don't understand about the last paragraph, why do I need to add in this part of the documentation?

Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: not in this part. sorry! It was just to remind that there was a special section elsewhere about symbol style and if focus on font marker (or anything like this) is needed, could be better to add it there.
So no need to touch that section in this PR

**Comment by SrNetoChan** [comment]
LGTM +1

**Pull request merged**